linchpin/provision/filter_plugins/get_pod_status.py

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env python
+import sys
+
+def get_pod_status(pod_list, res_def_out):
+    # this filter runs only when the kind is "Pod"
+    pod_status = pod_list.get("resources", [])
+    container_names = [x["name"] for x in res_def_out["result"]["spec"]["containers"]]
+    pod_stats = {}
+    # filters out all the pods data w.r.to containers 
+    # provisioned in pods
+    for x in pod_status:
+        if x["metadata"]["name"] in container_names:
+            pod_data = x
+            pod_data["status"] = x["status"]["phase"]
+            pod_stats[x["metadata"]["name"]] = pod_data
+    # checks for the status of the container and returns Failure value
+    # when any container is not in Running state
+    for name in container_names:
+        if pod_stats[name]["status"] != "Running":
```

Comment:
is "running the only successful status"? I am not sure abt this, will have to check docs

Reply:
@skatlapa 
I believe for containers running is only success state